modules/system/ServiceProvider.php

```diff
@@ -104,6 +104,21 @@ public function boot()
             Config::set('session.same_site', 'Lax');
         }
 
+        /*
+         * Set a default cookie prefix
+         * Spec https://tools.ietf.org/html/draft-ietf-httpbis-cookie-prefixes-00#section-3
+         */
+        if ((!preg_match('/^__/', Config::get('session.cookie'))) && (Config::get('session.secure') == true) && (Request::secure())) {
+            // Set the host prefix
+            if ((strtolower(Config::get('session.cookie_prefix')) === 'host') && (Config::get('session.path') === '/')) {
+                Config::set('session.cookie', '__Host-'.Config::get('session.cookie'));
+
+            // Set the secure prefix to everything else as the default
+            } elseif (strtolower(Config::get('session.cookie_prefix')) !== 'none') {
+                Config::set('session.cookie', '__Secure-'.Config::get('session.cookie'));
+            }
```

Comment:
@ayumi-cloud We'll need to set the other config values to match the prefix's requirements as well, otherwise, there's no point to this and may result in people using configuration combinations that simply won't work.

```suggestion
            if ((strtolower(Config::get('session.cookie_prefix')) === 'host')) {
                // Set the host prefix
                Config::set('session.cookie', '__Host-' . Config::get('session.cookie'));
                Config::set('session.secure', true);
                Config::set('session.path', '/');
            } elseif (strtolower(Config::get('session.cookie_prefix')) === 'secure') {
                // Set the secure prefix
                Config::set('session.cookie', '__Secure-'.Config::get('session.cookie'));
                Config::set('session.secure', true);
            }
```

Reply:
@bennothommo I'm just looking at your code suggestion change, it looks all good. 

I noticed a couple of errors.

### Issue 1

Except that the `host` must **not** run a subdomain - code line 112 needs changing.

The `__Host-` can't be set on a subdomain!

That's why I was checking the `Config::get('session.path') === '/'` - which probably isn't the best method in hindsight.

This code line needs changing:

```
if (strtolower(Config::get('session.cookie_prefix')) === 'host') {
```

To also check that the cookie is running only on the host domain and **not a subdomain**.

### Issue 2

Code line 117, I set it to `not equal to none` and not `equal to secure` on **purpose!**

The cms should add the `secure` to all the session cookies as the **default**.

> I should mention, before opening the issue and writing this pr. We scraped a list of 30,000 October websites and checked their cookies! Over 99% of the websites didn't use a cookie prefix and this was the main reason why we decided to open this issue. We decided in the office that the best course of action was for the cms to automatically add the cookie prefix into the session cookie name!

If a users adds this:

```
'cookie_prefix' => 'acme',
```

The cms will still automatically add the `secure` prefix.

The only way to **opt-out** of not adding the `secure` cookie prefix is to add the value `none`.

Hope that makes sense.

Reply:
Ignore issue 1.

Only issue 2 is valid.

Reply:
Please give feedback on commit: https://github.com/octobercms/october/pull/5468/commits/89fcc1922293998c22fe60abfa8c3b8c2969201b

I did everything you suggested, except using `!== 'none` instead of `=== 'secure'`